mempool/clist_mempool.go

```diff
@@ -676,7 +676,6 @@ func (mem *CListMempool) recheckTxs() {
 		})
 		if err != nil {
 			mem.logger.Error("recheckTx", err, "err")
-			return
 		}
```

Comment:
Wouldn't it be better to nullify `recheckCursor` (and maybe `recheckEnd`) and still make an early return? After a transaction check has failed, is it wasted work to check the remaining transactions?

We also may have a "too much object state" problem where this method modifies mempool object members that may be used by other methods outside of a single recheck run. This could be despaghettified by factoring the recheck state off to a separate object that is passed to the global callback instead of, or alongside with, the mempool object.

Reply:
This may work for the local client but not for the socket client. The first one makes sync calls to the app and calls the callback after each response. The second sends all the requests and then handles the responses asynchronously.